drivers/regulator/regulator_mpm54304.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2024 Vogl Electronic GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT mps_mpm54304
+
+#include <errno.h>
+#include <stdint.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mps_mpm54304, CONFIG_REGULATOR_LOG_LEVEL);
+
+#define MPM54304_REG_EN		0x0CU
+#define MPM54304_REG_VENDOR_ID	0x13U
+
+#define MPM54304_buck1_EN_MASK BIT(7)
+#define MPM54304_buck2_EN_MASK BIT(6)
+#define MPM54304_buck3_EN_MASK BIT(5)
+#define MPM54304_buck4_EN_MASK BIT(4)
+
+struct regulator_mpm54304_config {
+	struct regulator_common_config common;
+	struct i2c_dt_spec bus;
+	uint8_t enable_mask;
+};
+
+struct regulator_mpm54304_data {
+	struct regulator_common_data common;
+};
+
+static int regulator_mpm54304_enable(const struct device *dev)
+{
+	const struct regulator_mpm54304_config *config = dev->config;
+
+	return i2c_reg_update_byte_dt(&config->bus, MPM54304_REG_EN,
+				      config->enable_mask, UINT8_MAX);
```

Comment:
still doesn't give a clue on why UINT8_MAX is needed, eg, many enable regs require just a `1`.

Reply:
I want to set all bits of the register `MPM54304_REG_EN` which are also set in the mask  `config->enable_mask`. I could do following, but it might be still confusing.
```
return i2c_reg_update_byte_dt(&config->bus, MPM54304_REG_EN, config->enable_mask, config->enable_mask);
```

Reply:
yes, using enable_mask is the correct way, even if the value gets masked, otherwise it's confusing.